File: Java Implementation/toxic.java
import com.google.cloud.language.v1.*;
import net.dv8tion.jda.api.*;
import net.dv8tion.jda.api.entities.*;
import net.dv8tion.jda.api.events.message.*;
import net.dv8tion.jda.api.hooks.ListenerAdapter;

public class ToxicBot extends ListenerAdapter {
    private final String prefix = "!toxic_bot";
    private final LanguageServiceClient languageClient;
    private final JDA discord;

    public ToxicBot(String discordToken) {
        this.discord = new JDABuilder(discordToken).addEventListeners(this).build();
        this.languageClient = LanguageServiceClient.create();
    }

    @Override
    public void onMessageReceived(MessageReceivedEvent event) {
        if (!event.getAuthor().isBot() && event.getMessage().getContentRaw().startsWith(prefix)) {
            String userMessage = event.getMessage().getContentRaw().substring(prefix.length()).trim();

            // perspective ai
            Document document = Document.newBuilder().setContent(userMessage).setType(Document.Type.PLAIN_TEXT).build();
            AnalyzeSentimentResponse response = languageClient.analyzeSentiment(AnalyzeSentimentRequest.newBuilder().setDocument(document).setEncodingType(EncodingType.UTF8).build());

            // toxic %
            double toxicityPercentage = Math.round(response.getDocumentSentiment().getScore() * 100 * 100.0) / 100.0;

            MessageChannel channel = event.getChannel();
            channel.sendMessage("The toxicity percentage of your message is " + toxicityPercentage + "%.").queue();
        }
    }

    public static void main(String[] args) {
        String discordToken = "DISCORD_BOT_TOKEN";
        new ToxicBot(discordToken);
    }
}

File: Python Implementation/toxic.py
import discord
from google.cloud import language_v1 #will be using Google's Perspective AI

client = discord.Client()

@client.event
async def on_ready():
    print('Logged in through {0.user}'.format(client))

@client.event
async def on_message(message):
    if message.content.startswith('!toxic_bot'):
        user_message = message.content.split('!toxic_bot')[1].strip()

        client = language_v1.LanguageServiceClient()
        document = language_v1.Document(content=user_message, type_=language_v1.Document.Type.PLAIN_TEXT)
        response = client.analyze_sentiment(request={'document': document, 'features': {'extract_document_sentiment': True, 'extract_entity_sentiment': False, 'extract_syntax': False}})

        # toxic %
        toxicity_percentage = round(response.document_sentiment.score * 100, 2)

        await message.channel.send(f'The toxicity percentage of your message is {toxicity_percentage}%.')

client.run('DISCORD_BOT_TOKEN')
